pubsub/mqtt/mqtt.go

```diff
@@ -281,8 +281,9 @@ func (m *mqttPubSub) onMessage(ctx context.Context) func(client mqtt.Client, mqt
 		}()
 
 		msg := pubsub.NewMessage{
-			Topic: mqttMsg.Topic(),
-			Data:  mqttMsg.Payload(),
+			Topic:    mqttMsg.Topic(),
+			Data:     mqttMsg.Payload(),
+			Metadata: map[string]string{"retained": fmt.Sprint(mqttMsg.Retained())},
```

Comment:
Can you use strconv to convert to string? Otherwise, LGTM

Reply:
thanks for the suggestion! I changed it to strconv